dts/arm/armv6-m.dtsi

```diff
@@ -22,6 +22,11 @@
 			reg = <0xe000e010 0x10>;
 			status = "disabled";
 		};
+
+		mpu: mpu@e000ed90 {
```

Comment:
does this really make sense here, since not all arm soc's have MPUs.

Reply:
The MPU registers can always be accessed (some of them will be RAZ/WI, of course, if the MPU is not present). So this does not harm, IMHO, as we don't really define the "regions" property, here.  The "regions" property and the corresponding DT_ macro are, anyway,  defined in SOC/Board DTS fixup. 

I don't see a better place to define the MPU node, actually. We do the same for NVIC. And the actual "prio-bits" property is defined in the SOC level.

Reply:
But I feel like all the SoCs have NVIC which is why we do it, and if one doesn't it should delete the node.

I'd rather see we add MPU nodes to the SoCs that have them instead of at this level.